src/gluonts/core/serde/_base.py

```diff
@@ -309,13 +309,13 @@ def decode(r: Any) -> Any:
     """
 
     # structural recursion over the possible shapes of r
-    if type(r) == dict and "__kind__" in r:
+    if isinstance(r, dict) and "__kind__" in r:
         kind = r["__kind__"]
         cls = cast(Any, locate(r["class"]))
 
         assert cls is not None, f"Can not locate {r['class']}."
 
-        if kind == Kind.Type:
+        if kind.value == Kind.Type.value:
```

Comment:
What happens if you reset only this line to what it was before?

Reply:
In the original state this line was not passing the style checks.

Reply:
I think this line had no issues with the checks. It was only the `type() == ` lines.